src/ExpressionEvaluator/Core/Source/ExpressionCompiler/DynamicFlagsCustomTypeInfo.cs

```diff
@@ -19,26 +19,18 @@ private DynamicFlagsCustomTypeInfo(ReadOnlyCollection<byte> bytes)
             _bytes = bytes;
         }
 
-        public DynamicFlagsCustomTypeInfo(Guid payloadTypeId, ReadOnlyCollection<byte> payload)
-            : this(payloadTypeId == PayloadTypeId ? payload : null)
+        private DynamicFlagsCustomTypeInfo(ArrayBuilder<bool> dynamicFlags, int startIndex)
         {
-        }
-
-        public DynamicFlagsCustomTypeInfo(DkmClrCustomTypeInfo typeInfo)
-            : this(typeInfo != null && typeInfo.PayloadTypeId == PayloadTypeId ? typeInfo.Payload : null)
-        {
-        }
+            Debug.Assert(startIndex >= 0);
 
-        internal DynamicFlagsCustomTypeInfo(bool[] dynamicFlags)
-        {
-            if (dynamicFlags == null || dynamicFlags.Length == 0)
+            if (dynamicFlags == null || dynamicFlags.Count <= startIndex)
```

Comment:
Rather than allowing null ArrayBuilder<bool>, consider asserting arg is non-null and having the (one?) caller use default(DynamicFlagsCustomTypeInfo) instead.